src/tests/integration/monitor-tabs.test.ts

```diff
@@ -0,0 +1,200 @@
+import { expect } from 'chai'
+import { createMocks } from './mocks'
+import { runBackground } from './steps/run-background'
+
+describe.only('monitor tabs', () => {
+  const mocks = createMocks()
+
+  runBackground(mocks, { alreadyAuthenticated: true })
+
+  describe('chrome.tabs.onCreated', () => {
+    it("adds the newly created tab to the store's state", () => {
+      expect(mocks.chrome.tabs.onCreated.addListener).to.have.callCount(1)
+      const [callback] = mocks.chrome.tabs.onCreated.addListener.firstCall.args
+      callback({
+        id: 18,
+        windowId: 1,
+        active: false,
+        url: 'https://new.com/abc',
+      })
+      const state = mocks.getState()
+      const tab = state.tabs.get(18)!
+      expect(tab.id).to.equal(18)
+      expect(tab.windowId).to.equal(1)
+      expect(tab.domain).to.equal('new.com')
+      expect(tab.url).to.equal('https://new.com/abc')
+    })
+  })
+
+  describe('chrome.tabs.onRemoved', () => {
+    it("removes a tab from the store's state", () => {
+      const [callbackOnRemove] = mocks.chrome.tabs.onRemoved.addListener.firstCall.args
+
+      const tabId = 11
+      const removeInfo = {
+        id: 11,
+        windowId: 1,
+        active: false,
+        url: 'https://foo.com/abc',
+      }
+      callbackOnRemove(tabId, removeInfo)
+
+      const state = mocks.getState()
+      const tabs = state.tabs
+
+      expect(tabs.get(tabId)).to.equal(undefined)
+    })
+  })
+
+  describe('chrome.tabs.onUpdated', () => {
+    it("updates a tab change to store's state", () => {
+      const [callbackOnUpdated] = mocks.chrome.tabs.onUpdated.addListener.firstCall.args
+
+      const tabId = 16
+      const changeInfo = {
+        id: 16,
+        windowId: 3,
+        active: false,
+        url: 'https://changeinfo.com/abc',
+      }
+
+      callbackOnUpdated(tabId, changeInfo)
+
+      const state = mocks.getState()
+      const tab = state.tabs.get(tabId)!
+
+      expect(tab.id).to.equal(tabId)
+      expect(tab.windowId).to.equal(changeInfo.windowId)
+      expect(tab.domain).to.equal('changeinfo.com')
+      expect(tab.url).to.equal('https://changeinfo.com/abc')
+    })
+  })
+
+  describe('chrome.tabs.onAttached', () => {
+    it("Attaches a tab to another window and updates the store's state", () => {
+      const [callbackOnAttached] = mocks.chrome.tabs.onAttached.addListener.firstCall.args
+
+      const tabId = 15
+      const attachInfo = {
+        id: 15,
+        windowId: 2,
+        newWindowId: 3,
+        active: false,
+        url: 'https://slam.com/abc',
+      }
+
+      callbackOnAttached(tabId, attachInfo)
+
+      const state = mocks.getState()
+      const tab = state.tabs.get(tabId)!
+
+      expect(tab.id).to.equal(attachInfo.id)
+      expect(tab.windowId).to.equal(attachInfo.newWindowId)
+      expect(tab.domain).to.equal('slam.com')
```

Comment:
This is misleading as it appears to indicate that the tab domain came from those sent in the callback, which they didn't, let's remove these.

Reply:
Refactor as discussed.

Reply:
Remove assertion syntax for domain.